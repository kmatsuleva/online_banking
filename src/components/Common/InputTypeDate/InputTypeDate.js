import React from "react";
import PropTypes from 'prop-types';     

const InputTypeDate = ({  min, max, style }) => (
    <input type="date" min={min} max = {max} style={style}/>  
);


InputTypeDate.propTypes = {
    min: PropTypes.string,
    max: PropTypes.string,
    style: PropTypes.object  
};

InputTypeDate.defaultProps = {
    style: {
        border: "1px solid #d2d2d2",
        padding: "8px 10px",
        marginTop: 8,
        marginBottom: 15,
        borderRadius: 3,
        height: 23,
        fontSize: 15,
        color: "#4a4a4a",
        boxShadow: "inset 1px 1px 2px rgba(0, 0, 0, 0.2)",
        width: 170,
        fontFamily: "Roboto, Arial, sans-serif"
    },

    min: "2018-01-01",
    max: "2022-01-01"
}



export default InputTypeDate;


