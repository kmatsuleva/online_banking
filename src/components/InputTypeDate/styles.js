const styles = {
    inputDate: {
        border: "1px solid #d2d2d2",
        padding: "8px 10px",
        marginTop: 8,
        marginBottom: 15,
        borderRadius: 3,
        height: 23,
        fontSize: 15,
        color: "#4a4a4a",
        boxShadow: "inset 1px 1px 2px rgba(0, 0, 0, 0.2)",
        width: 170,
        fontFamily: "Roboto, Arial, sans-serif"
    }
}

export default styles;