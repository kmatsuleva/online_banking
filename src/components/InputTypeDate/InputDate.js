import React from "react";
import styles from "./styles"

const InputDate = () => (
      <input type="date" min="2013-12-31"
            style = {styles.inputDate}/>
);

export default InputDate;
   