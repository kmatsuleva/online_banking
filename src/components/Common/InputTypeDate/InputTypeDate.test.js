
import React from 'react'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import InputTypeDate from "./InputTypeDate"  

Enzyme.configure({ adapter: new Adapter() })


describe("< InputTypeDate />", () => {

    it('renders without crashing', () => {
        shallow(<InputTypeDate />);
    });

    it('should render correctly', () => {
        const output = shallow(
            <InputTypeDate min="2018-01-01" max="2022-08-09"/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});