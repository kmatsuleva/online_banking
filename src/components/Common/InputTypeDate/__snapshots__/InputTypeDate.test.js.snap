// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< InputTypeDate /> should render correctly 1`] = `
<input
  max="2022-08-09"
  min="2018-01-01"
  style={
    Object {
      "border": "1px solid #d2d2d2",
      "borderRadius": 3,
      "boxShadow": "inset 1px 1px 2px rgba(0, 0, 0, 0.2)",
      "color": "#4a4a4a",
      "fontFamily": "Roboto, Arial, sans-serif",
      "fontSize": 15,
      "height": 23,
      "marginBottom": 15,
      "marginTop": 8,
      "padding": "8px 10px",
      "width": 170,
    }
  }
  type="date"
/>
`;
